meta/reflect.go

```diff
@@ -47,6 +47,14 @@ func HasPointers(any interface{}) bool {
 	return ElementType(any).Kind() == reflect.Ptr
 }
 
+func StructFieldValue(any interface{}, name string) any {
+	rv := reflect.ValueOf(any)
+	if rv.Kind() == reflect.Pointer {
+		rv = rv.Elem()
+	}
+	return rv.FieldByName(name).Interface()
```

Comment:
ChatGPT spotted an issue
```
If the field name does not exist or is not exported (i.e., its name doesn’t start with an uppercase letter), FieldByName will return a zero reflect.Value, and calling .Interface() on it will cause a runtime panic.
```

Reply:
yeah makes sense to me, i think a panic would be appropriate there since trying to use this library with unexported fields would be a "programmer error" where it makes sense to panic

Reply:
Normally, panic in Go is the last resort, when everything went wrong, unhandled unrecoverable case. "programmer error" should still be delivered as an error. It's a good practice to have a proper error handling. Cause you can tell more about what went wrong and you give the interface user a chance to recover. Which very common among util packages, also the std ones. It's not common to handle panics.

Reply:
yeah avoiding panics is a good rule of thumb but it's only a rule of thumb, not gospel imo. in this case we only panic if the programmer has forgotten to export a field. it's not something than can happen dynamically at runtime unexpectedly

also, there is no way we can gracefully handle this error. if there is an unexported field this library is useless and so is the application using it. i think it should be unrecoverable

also there are lots of cases in the stdlib of "panic from programmer error"  like strings.NewReplacer, and http.ServeMux.Handle come to mind but there are lots. generally not ones you want to recover from

Reply:
> in this case we only panic if the programmer has forgotten to export a field

Or if the field does not exist.

> it's not something than can happen dynamically at runtime unexpectedly

Why not? The field name is passed as a string. Compiler is not validating anything here at build time. So I can easily pass a value here in the runtime that will not exist in the struct.

> also, there is no way we can gracefully handle this error. if there is an unexported field this library is useless and so is the application using it. i think it should be unrecoverable

Can't it just return `nil` when there is no field? `map` does not panic when the user tries to access non-existent keys.

> generally not ones you want to recover from

Yeah, it's probably debatable what should be recoverable. My rule of thumb is to avoid panics whenever possible. I had an incident in the past when it brought down a vital service breaking checkout on the website. One developer used it as an error issuing mechanism. The panic was not caught and killed the main process.